Content.Server/Shadowkin/ShadowkinSystem.cs

```diff
@@ -0,0 +1,190 @@
+using Content.Shared.Examine;
+using Content.Shared.Abilities.Psionics;
+using Content.Shared.Humanoid;
+using Content.Shared.Psionics;
+using Content.Shared.Bed.Sleep;
+using Content.Shared.Damage.Components;
+using Content.Shared.Damage.Systems;
+using Content.Shared.Damage;
+using Content.Shared.FixedPoint;
+using Content.Shared.Shadowkin;
+using Content.Shared.Rejuvenate;
+using Content.Shared.Alert;
+using Content.Shared.Rounding;
+using Content.Shared.Actions;
+using Robust.Shared.Prototypes;
+using Content.Server.Abilities.Psionics;
+
+namespace Content.Server.Shadowkin;
+public sealed class ShadowkinSystem : EntitySystem
+{
+    [Dependency] private readonly StaminaSystem _stamina = default!;
+    [Dependency] private readonly DamageableSystem _damageable = default!;
+    [Dependency] private readonly PsionicAbilitiesSystem _psionicAbilitiesSystem = default!;
+    [Dependency] private readonly AlertsSystem _alerts = default!;
+    [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    public const string ShadowkinSleepActionId = "ShadowkinActionSleep";
```

Comment:
This should be a datafield on the ShadowkinComponent instead of a public constant.